File: Compiler/TranslatorTests/TestProjects/18/TestIssue379.cs
using Bridge;

namespace TestIssue379
{
    [ObjectLiteral(DefaultValueMode.Ignore)]
    public class DataIgnore
    {
        public DataIgnore()
        {
        }

        public DataIgnore(DefaultValueMode mode)
        {

        }

        public int Int1;
        public int Int2 = 2;
        public string Str3;
        public string Str4 = "Str4";
        public int? IntNull5;
        public int? IntNull6 = 6;
        public decimal Decimal7;
        public decimal Decimal8 = 8;
    }


    [ObjectLiteral(DefaultValueMode.DefaultValue)]
    public class DataDefaultValue
    {
        public DataDefaultValue()
        {
        }

        public DataDefaultValue(DefaultValueMode mode)
        {

        }

        public int Int1;
        public int Int2 = 2;
        public string Str3;
        public string Str4 = "Str4";
        public int? IntNull5;
        public int? IntNull6 = 6;
        public decimal Decimal7;
        public decimal Decimal8 = 8;
    }

    [ObjectLiteral(DefaultValueMode.Initializer)]
    public class DataInitializer
    {
        public DataInitializer()
        {
        }

        public DataInitializer(DefaultValueMode mode)
        {

        }

        public int Int1;
        public int Int2 = 2;
        public string Str3;
        public string Str4 = "Str4";
        public int? IntNull5;
        public int? IntNull6 = 6;
        public decimal Decimal7;
        public decimal Decimal8 = 8;
    }


    public class Tests
    {
        public void TestDataIgnore()
        {
            var d1 = new DataIgnore();
            var d2 = new DataIgnore() { Int1 = 1, Int2 = 22, Str3 = "3", Str4 = "Str44", IntNull5 = 5, IntNull6 = 66, Decimal7 = 7, Decimal8 = 88};

            var d3 = new DataIgnore(DefaultValueMode.Ignore);
            var d4 = new DataIgnore(DefaultValueMode.Ignore) { Int1 = 1, Int2 = 22, Str3 = "3", Str4 = "Str44", IntNull5 = 5, IntNull6 = 66, Decimal7 = 7, Decimal8 = 88 };

            var d5 = new DataIgnore(DefaultValueMode.DefaultValue);
            var d6 = new DataIgnore(DefaultValueMode.DefaultValue) { Int1 = 1, Int2 = 22, Str3 = "3", Str4 = "Str44", IntNull5 = 5, IntNull6 = 66, Decimal7 = 7, Decimal8 = 88 };

            var d7 = new DataIgnore(DefaultValueMode.Initializer);
            var d8 = new DataIgnore(DefaultValueMode.Initializer) { Int1 = 1, Int2 = 22, Str3 = "3", Str4 = "Str44", IntNull5 = 5, IntNull6 = 66, Decimal7 = 7, Decimal8 = 88 };
        }

        public void TestDataDefaultValue()
        {
            var d1 = new DataDefaultValue();
            var d2 = new DataDefaultValue() { Int1 = 1, Int2 = 22, Str3 = "3", Str4 = "Str44", IntNull5 = 5, IntNull6 = 66, Decimal7 = 7, Decimal8 = 88 };

            var d3 = new DataDefaultValue(DefaultValueMode.Ignore);
            var d4 = new DataDefaultValue(DefaultValueMode.Ignore) { Int1 = 1, Int2 = 22, Str3 = "3", Str4 = "Str44", IntNull5 = 5, IntNull6 = 66, Decimal7 = 7, Decimal8 = 88 };

            var d5 = new DataDefaultValue(DefaultValueMode.DefaultValue);
            var d6 = new DataDefaultValue(DefaultValueMode.DefaultValue) { Int1 = 1, Int2 = 22, Str3 = "3", Str4 = "Str44", IntNull5 = 5, IntNull6 = 66, Decimal7 = 7, Decimal8 = 88 };

            var d7 = new DataDefaultValue(DefaultValueMode.Initializer);
            var d8 = new DataDefaultValue(DefaultValueMode.Initializer) { Int1 = 1, Int2 = 22, Str3 = "3", Str4 = "Str44", IntNull5 = 5, IntNull6 = 66, Decimal7 = 7, Decimal8 = 88 };
        }

        public void TestDataInitializer()
        {
            var d1 = new DataInitializer();
            var d2 = new DataInitializer() { Int1 = 1, Int2 = 22, Str3 = "3", Str4 = "Str44", IntNull5 = 5, IntNull6 = 66, Decimal7 = 7, Decimal8 = 88 };

            var d3 = new DataInitializer(DefaultValueMode.Ignore);
            var d4 = new DataInitializer(DefaultValueMode.Ignore) { Int1 = 1, Int2 = 22, Str3 = "3", Str4 = "Str44", IntNull5 = 5, IntNull6 = 66, Decimal7 = 7, Decimal8 = 88 };

            var d5 = new DataInitializer(DefaultValueMode.DefaultValue);
            var d6 = new DataInitializer(DefaultValueMode.DefaultValue) { Int1 = 1, Int2 = 22, Str3 = "3", Str4 = "Str44", IntNull5 = 5, IntNull6 = 66, Decimal7 = 7, Decimal8 = 88 };

            var d7 = new DataInitializer(DefaultValueMode.Initializer);
            var d8 = new DataInitializer(DefaultValueMode.Initializer) { Int1 = 1, Int2 = 22, Str3 = "3", Str4 = "Str44", IntNull5 = 5, IntNull6 = 66, Decimal7 = 7, Decimal8 = 88 };
        }
    }
}

File: Compiler/TranslatorTests/TestProjects/18/Bridge/reference/testIssue379.js
/* global Bridge */

(function (globals) {
    "use strict";

    Bridge.define('TestIssue379.Tests', {
        testDataIgnore: function () {
            var d1 = {  };
            var d2 = { int1: 1, int2: 22, str3: "3", str4: "Str44", intNull5: 5, intNull6: 66, decimal7: Bridge.Decimal(7), decimal8: Bridge.Decimal(88) };
    
            var d3 = {  };
            var d4 = { int1: 1, int2: 22, str3: "3", str4: "Str44", intNull5: 5, intNull6: 66, decimal7: Bridge.Decimal(7), decimal8: Bridge.Decimal(88) };
    
            var d5 = { int1: 0, int2: 2, str3: null, str4: "Str4", intNull5: null, intNull6: 6, decimal7: Bridge.Decimal(0.0), decimal8: Bridge.Decimal(8) };
            var d6 = { int1: 1, int2: 22, str3: "3", str4: "Str44", intNull5: 5, intNull6: 66, decimal7: Bridge.Decimal(7), decimal8: Bridge.Decimal(88) };
    
            var d7 = { int2: 2, str4: "Str4", intNull6: 6, decimal8: Bridge.Decimal(8) };
            var d8 = { int1: 1, int2: 22, str3: "3", str4: "Str44", intNull5: 5, intNull6: 66, decimal7: Bridge.Decimal(7), decimal8: Bridge.Decimal(88) };
        },
        testDataDefaultValue: function () {
            var d1 = { int1: 0, int2: 2, str3: null, str4: "Str4", intNull5: null, intNull6: 6, decimal7: Bridge.Decimal(0.0), decimal8: Bridge.Decimal(8) };
            var d2 = { int1: 1, int2: 22, str3: "3", str4: "Str44", intNull5: 5, intNull6: 66, decimal7: Bridge.Decimal(7), decimal8: Bridge.Decimal(88) };
    
            var d3 = {  };
            var d4 = { int1: 1, int2: 22, str3: "3", str4: "Str44", intNull5: 5, intNull6: 66, decimal7: Bridge.Decimal(7), decimal8: Bridge.Decimal(88) };
    
            var d5 = { int1: 0, int2: 2, str3: null, str4: "Str4", intNull5: null, intNull6: 6, decimal7: Bridge.Decimal(0.0), decimal8: Bridge.Decimal(8) };
            var d6 = { int1: 1, int2: 22, str3: "3", str4: "Str44", intNull5: 5, intNull6: 66, decimal7: Bridge.Decimal(7), decimal8: Bridge.Decimal(88) };
    
            var d7 = { int2: 2, str4: "Str4", intNull6: 6, decimal8: Bridge.Decimal(8) };
            var d8 = { int1: 1, int2: 22, str3: "3", str4: "Str44", intNull5: 5, intNull6: 66, decimal7: Bridge.Decimal(7), decimal8: Bridge.Decimal(88) };
        },
        testDataInitializer: function () {
            var d1 = { int2: 2, str4: "Str4", intNull6: 6, decimal8: Bridge.Decimal(8) };
            var d2 = { int1: 1, int2: 22, str3: "3", str4: "Str44", intNull5: 5, intNull6: 66, decimal7: Bridge.Decimal(7), decimal8: Bridge.Decimal(88) };
    
            var d3 = {  };
            var d4 = { int1: 1, int2: 22, str3: "3", str4: "Str44", intNull5: 5, intNull6: 66, decimal7: Bridge.Decimal(7), decimal8: Bridge.Decimal(88) };
    
            var d5 = { int1: 0, int2: 2, str3: null, str4: "Str4", intNull5: null, intNull6: 6, decimal7: Bridge.Decimal(0.0), decimal8: Bridge.Decimal(8) };
            var d6 = { int1: 1, int2: 22, str3: "3", str4: "Str44", intNull5: 5, intNull6: 66, decimal7: Bridge.Decimal(7), decimal8: Bridge.Decimal(88) };
    
            var d7 = { int2: 2, str4: "Str4", intNull6: 6, decimal8: Bridge.Decimal(8) };
            var d8 = { int1: 1, int2: 22, str3: "3", str4: "Str44", intNull5: 5, intNull6: 66, decimal7: Bridge.Decimal(7), decimal8: Bridge.Decimal(88) };
        }
    });
    
    
    
    Bridge.init();
})(this);
